internal/install/recipe_installer_test.go

```diff
@@ -892,13 +890,116 @@ func TestInstall_GuidReport(t *testing.T) {
 	require.Equal(t, 1, statusReporters[0].(*execution.MockStatusReporter).RecipeSkippedCallCount)
 	require.Equal(t, rv.ValidateVal, statusReporters[0].(*execution.MockStatusReporter).RecipeGUID[types.InfraAgentRecipeName])
 	require.Equal(t, rv.ValidateVal, statusReporters[0].(*execution.MockStatusReporter).RecipeGUID[testRecipeName])
-	require.Equal(t, status.CLIVersion, "v0.0.1")
+	require.Equal(t, status.CLIVersion, cli.Version())
 	require.Equal(t, 3, len(statusReporters[0].(*execution.MockStatusReporter).Durations))
 	for _, duration := range statusReporters[0].(*execution.MockStatusReporter).Durations {
 		require.Less(t, int64(0), duration)
 	}
 }
 
+func TestInstall_ShouldDetectRecipe(t *testing.T) {
+	os.Setenv("NEW_RELIC_ACCOUNT_ID", "12345")
+	ic := types.InstallerContext{}
+	statusReporters = []execution.StatusSubscriber{execution.NewMockStatusReporter()}
+	status = execution.NewInstallStatus(statusReporters, execution.NewPlatformLinkGenerator())
+	rf := recipes.NewRecipeFilterRunner(ic, status)
+	f = recipes.NewMockRecipeFetcher()
+
+	recipe := types.OpenInstallationRecipe{
+		Name:           testRecipeName,
+		ValidationNRQL: "testNrql",
+	}
+
+	f.FetchRecipesVal = []types.OpenInstallationRecipe{recipe}
+
+	v = validation.NewMockRecipeValidator()
+
+	i := RecipeInstaller{ic, d, l, mv, f, e, v, ff, status, p, pi, sp, lkf, cv, rvp, rf, av}
+
+	err := i.Install()
+
+	require.NoError(t, err)
+	require.Equal(t, 1, statusReporters[0].(*execution.MockStatusReporter).RecipeDetectedCallCount)
+}
+
+func TestInstall_ShouldDetectMultipleRecipes(t *testing.T) {
+	os.Setenv("NEW_RELIC_ACCOUNT_ID", "12345")
+	ic := types.InstallerContext{}
+	statusReporters = []execution.StatusSubscriber{execution.NewMockStatusReporter()}
+	status = execution.NewInstallStatus(statusReporters, execution.NewPlatformLinkGenerator())
+	rf := recipes.NewRecipeFilterRunner(ic, status)
+	f = recipes.NewMockRecipeFetcher()
+
+	f.FetchRecipesVal = []types.OpenInstallationRecipe{
+		// Should be detected
+		{
+			Name:           "compatible-recipe-1",
+			ValidationNRQL: "testNrql",
+		},
+		// Should be detected
+		{
+			Name:           "compatible-recipe-2",
+			ValidationNRQL: "testNrql",
+		},
+	}
+
+	v = validation.NewMockRecipeValidator()
+
+	i := RecipeInstaller{ic, d, l, mv, f, e, v, ff, status, p, pi, sp, lkf, cv, rvp, rf, av}
+
+	err := i.Install()
+
+	require.NoError(t, err)
+	require.Equal(t, 2, statusReporters[0].(*execution.MockStatusReporter).RecipeDetectedCallCount)
+}
+
+func TestInstall_ShouldNotDetectRecipeWithProcessMatch(t *testing.T) {
```

Comment:
I think this test is misnamed, it is validating that we detect when process match and scripts indicate the specific value of 132.

Reply:
We also need the flip side of this test, a test that validate we don't detect when process match, and the script returns an exit code that is NOT 132.

Reply:
This test is difficult to name. It's testing that a process match was found, but the preinstall check failed with exit code 132, meaning we did _not_ detect compatibility. But yes we need the flip side of this too.

Reply:
@sanderblue actually this test is incorrect, check back the AC in the story. If process matches, and script is present with an exit code of 132, the Detected event should be fired up.